std/algorithm/mutation.d

```diff
@@ -562,6 +563,13 @@ void fill(Range, Value)(Range range, Value value)
     }
 }
 
+/// ditto
+void fill(A, Value)(auto ref A array, auto ref Value value)
+if (is(typeof((){array[] = value;})) && isSomeChar!Value && isMutable!Value)
```

Comment:
Btw I think `typeof` already checks for mutability and correctness of `opIndexAssign`, so I am not sure whether `isMutable!Value` is an unnecessary restriction. Consider this short snippet for example:

```d
    import std.stdio;
    char[] chars = ['.', '.'];
    pragma(msg, is(typeof(chars[] = 'a'))); // true
    const b = 'b';
    pragma(msg, is(typeof(chars[] = b))); // true
    chars[] = b;
    writeln(chars); // bb
    pragma(msg, isMutable!(typeof(b))); // false
```